models/m_audit.erl

```diff
@@ -131,6 +135,44 @@ user_agent_id(UserAgent, Context) ->
     Transaction = fun() -> z_db:transaction(F, Context) end,
     z_depcache:memo(Transaction, {user_agent_id, BUA}, ?MAXAGE_UA_STRING, Context).
 
+% Remove entries older than 2 years.
+periodic_cleanup(Context) ->
+    z_db:q("delete from audit where id in (select id from audit where created < now() - interval '2 years' limit 10000)", Context, 300000).
+
+
+% Update all records in the audit table
+update_audit_table(Context) ->
+    io:fwrite(standard_error, "Updating audit table", []),
+    update_audit_table1(Context).
+
+update_audit_table1(Context) ->
+    case props_json_update(Context) of
+        {ok, N} when N > 0 ->
+            io:fwrite(standard_error, ".", []),
+            update_audit_table1(Context);
+        {ok, 0} ->
+            io:fwrite(standard_error, ".~n", []),
+            done
```

Comment:
Maybe these cases could be reversed, so the guard can be dropped and is therefore simplified?

Reply:
Could do that, but for a single use function I wanted to have the "normal" case in front of the exceptional end case.